File: WEEK1/DAY6/Prog4/P4.java
import java.util.*;
public class P4 {
    public static void main(String[] args) {
        int N;
        Scanner sc=new Scanner(System.in);
        System.out.println("Find largest digit in N");
        System.out.print("Enter N: ");
        N=sc.nextInt();
        int n=N;
        int dig;
        int m=-1;
        while(n>0){
            dig=n%10;
            if(m<dig){
                m=dig;
            }
            n/=10;
        }
        System.out.println("The largest digit in "+N+" is "+m);
        sc.close();
    }
}

File: WEEK1/DAY6/Prog4/P4.c
#include <stdio.h>
int main(){
    int N;
    printf("Find largest digit in N\n");
    printf("Enter N: ");
    scanf("%d",&N);
    int m=-1;
    int n=N;
    int dig;
    while(n>0){
        dig=n%10;
        if(m<dig){
            m=dig;
        }
        n/=10;
    }
    printf("The largest digit in %d is %d\n",N,m);
}